Tests are failing at CircleCI

Tests are failing with an error:

```js
yarn run v1.9.4
warning ../../../package.json: No license field
$ BABEL_ENV=tests mocha --compilers js:babel-core/register --require tests/index.js tests/**
module.js:550
    throw err;
    ^

Error: Cannot find module './config/env.json'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/circleci/repo/packages/core/botpress/tests/bot.js:14:19)
    at Module._compile (module.js:653:30)
    at loader (/home/circleci/repo/packages/core/botpress/node_modules/babel-register/lib/node.js:144:5)
    at Object.require.extensions.(anonymous function) [as .js] (/home/circleci/repo/packages/core/botpress/node_modules/babel-register/lib/node.js:154:7)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at /home/circleci/repo/packages/core/botpress/node_modules/mocha/lib/mocha.js:231:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (/home/circleci/repo/packages/core/botpress/node_modules/mocha/lib/mocha.js:228:14)
    at Mocha.run (/home/circleci/repo/packages/core/botpress/node_modules/mocha/lib/mocha.js:514:10)
    at Object.<anonymous> (/home/circleci/repo/packages/core/botpress/node_modules/mocha/bin/_mocha:480:18)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
    at startup (bootstrap_node.js:204:16)
    at bootstrap_node.js:625:3
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Exited with code 1
```

I believe this started after https://github.com/botpress/botpress/pull/897.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -28,15 +28,6 @@ jobs:
           name: Install dependencies
           command: cd ~/repo/packages/core/botpress && yarn --force --build-from-source
 
-      - run:
-          name: Install ngrok
-          comand: |
-            wget https://s3.amazonaws.com/build-tools-prd/ngrok.zip
-            unzip ngrok.zip
-            ./ngrok http -subdomain=$CIRCLE_SHA1 3000
-            BOTPRESS_URL=https://$CIRCLE_SHA1.ngrok.com
-          background: true
-
       - save_cache:
           paths:
             - ~/repo/packages/core/botpress/node_modules

--- a/compile-link.sh
+++ b/compile-link.sh
@@ -1,3 +1,0 @@
-cd `dirname $0`
-./node_modules/.bin/lerna bootstrap
-./node_modules/.bin/lerna run compile

--- a/packages/core/botpress/src/cli/templates/init-default/src/custom.js
+++ b/packages/core/botpress/src/cli/templates/init-default/src/custom.js
@@ -18,7 +18,12 @@ async function registerActions(bp) {
   bp.dialogEngine.registerActionMetadataProvider(action => metadata[action])
 
   for (const actionFile of actionFiles) {
-    const docs = await jsdoc.explain({ files: path.join(__dirname, actionFile) })
+    // TODO: fails with "spawn node ENOENT" error on CircleCI otherwise
+    let docs = []
+    if (process.env.NODE_ENV !== 'test') {
+      docs = await jsdoc.explain({ files: path.join(__dirname, actionFile) })
+    }
+
     const actions = require(actionFile)
 
     for (const action of Object.keys(actions)) {

--- a/packages/core/botpress/tests/bot.js
+++ b/packages/core/botpress/tests/bot.js
@@ -1,109 +1,100 @@
+/* global before */
+
 import { expect } from 'chai'
 import path from 'path'
-import { existsSync } from 'fs'
-import { exec } from 'child_process'
+import { existsSync, symlinkSync } from 'fs'
+import { exec, execSync, spawnSync } from 'child_process'
 import moment from 'moment'
 import rimraf from 'rimraf'
 
 const bin = path.resolve(__dirname, '../bin/botpress')
 const botDir = path.resolve(__dirname, '../tmp-bot')
-const botConfig = path.resolve(__dirname, '../tmp-bot/config')
+const botConfig = path.resolve(__dirname, `${botDir}/config`)
 const rootDir = path.resolve(__dirname, '../../../../')
-const compileLinkBash = path.resolve(rootDir, './compile-link.sh')
 const configVar = require('./config/env.js')
 
-const botpressModules = [
-  `file:${path.resolve(rootDir, './packages/functionals/botpress-analytics')}`,
-  `file:${path.resolve(rootDir, './packages/functionals/botpress-audience')}`,
-  `file:${path.resolve(rootDir, './packages/functionals/botpress-broadcast')}`,
-  `file:${path.resolve(rootDir, './packages/functionals/botpress-hitl')}`,
-  `file:${path.resolve(rootDir, './packages/functionals/botpress-nlu')}`,
-  `file:${path.resolve(rootDir, './packages/functionals/botpress-qna')}`,
-  `file:${path.resolve(rootDir, './packages/functionals/botpress-scheduler')}`,
-  `file:${path.resolve(rootDir, './packages/functionals/botpress-terminal')}`,
-  `file:${path.resolve(rootDir, './packages/channels/botpress-channel-messenger')}`,
-  `file:${path.resolve(rootDir, './packages/channels/botpress-channel-slack')}`,
-  `file:${path.resolve(rootDir, './packages/channels/botpress-channel-telegram')}`,
-  `file:${path.resolve(rootDir, './packages/channels/botpress-channel-twilio')}`,
-  `file:${path.resolve(rootDir, './packages/channels/botpress-channel-microsoft')}`,
-  `file:${path.resolve(rootDir, './packages/channels/botpress-channel-web')}`,
-  `file:${path.resolve(rootDir, './packages/core/botpress-builtins')}`,
-  `file:${path.resolve(rootDir, './packages/core/botpress-util-roles')}`,
-  `file:${path.resolve(rootDir, './packages/skills/botpress-skill-choice')}`,
-  `file:${path.resolve(rootDir, './packages/core/botpress')}`
-]
+const botpressModules = {
+  botpress: './packages/core/botpress',
+  '@botpress/channel-messenger': './packages/channels/botpress-channel-messenger',
+  '@botpress/channel-slack': './packages/channels/botpress-channel-slack',
+  '@botpress/channel-telegram': './packages/channels/botpress-channel-telegram',
+  '@botpress/channel-twilio': './packages/channels/botpress-channel-twilio',
+  '@botpress/channel-microsoft': './packages/channels/botpress-channel-microsoft',
+  '@botpress/channel-web': './packages/channels/botpress-channel-web',
+  '@botpress/analytics': './packages/functionals/botpress-analytics',
+  '@botpress/audience': './packages/functionals/botpress-audience',
+  '@botpress/broadcast': './packages/functionals/botpress-broadcast',
+  '@botpress/hitl': './packages/functionals/botpress-hitl',
+  '@botpress/nlu': './packages/functionals/botpress-nlu',
+  '@botpress/qna': './packages/functionals/botpress-qna',
+  '@botpress/scheduler': './packages/functionals/botpress-scheduler',
+  '@botpress/terminal': './packages/functionals/botpress-terminal',
+  '@botpress/builtins': './packages/core/botpress-builtins',
+  '@botpress/util-roles': './packages/core/botpress-util-roles',
+  '@botpress/skill-choice': './packages/skills/botpress-skill-choice'
+}
 
 const LAUNCHED = /Bot launched/gi
 const TWENTY_MIN = 1200000
 
-describe('Create new bot', function() {
+describe('Running new bot', function() {
   this.timeout(TWENTY_MIN)
 
-  const startTime = moment()
-
   let hasError = false
-  const ifErrorSkipIt = () => expect(!!hasError).to.be.false
   const checkTaskStatus = err => {
     hasError = err
     expect(err).to.be.null
   }
 
-  // eslint-disable-next-line
   before(done => {
     if (existsSync(botDir)) {
-      rimraf(botDir, () => done())
-    } else {
-      done()
+      rimraf.sync(botDir)
     }
+    done()
   })
 
-  it('build all modules', done => {
-    ifErrorSkipIt()
+  beforeEach(done => {
+    // Tests here depend on previous steps so doesn't make sense to run further steps if previous failed
+    expect(!!hasError).to.be.false
+    done()
+  })
 
-    exec(`bash ${compileLinkBash}`, err => {
+  it('compile modules', done => {
+    const lerna = './node_modules/.bin/lerna'
+    spawnSync('yarn', { cwd: rootDir })
+    exec(`${lerna} bootstrap && ${lerna} run compile`, { cwd: rootDir }, err => {
       checkTaskStatus(err)
-
       done()
     })
   })
 
-  it('create', done => {
-    ifErrorSkipIt()
-
+  it('botpress init', done => {
     exec(`node ${bin} init ${botDir} -y`, err => {
       checkTaskStatus(err)
-
       done()
     })
   })
 
-  it('copy config', done => {
-    ifErrorSkipIt()
-
+  it('bot starts', done => {
+    // Create config-files
     rimraf.sync(botConfig)
-    exec(`cp -a ./tests/config ${botDir}`, err => {
-      checkTaskStatus(err)
+    spawnSync(`cp -a ./tests/config ${botDir}`)
 
-      done()
+    // Install dependencies and link to modules that we compiled
+    spawnSync('yarn', { cwd: botDir })
+    Object.entries(botpressModules).forEach(([name, modulePath]) => {
+      const linkPath = `${botDir}/node_modules/${name}`
+      if (existsSync(linkPath)) {
+        rimraf.sync(linkPath)
+      }
+      symlinkSync(path.resolve(rootDir, modulePath), linkPath)
     })
-  })
-
-  it('install modules', done => {
-    ifErrorSkipIt()
-
-    exec(`yarn add ${botpressModules.join(' ')}`, { cwd: botDir }, err => {
-      checkTaskStatus(err)
-
-      done()
-    })
-  })
-
-  it('run bot', done => {
-    ifErrorSkipIt()
 
     let lastDate = moment().toDate()
     let isLaunched = false
-    const botProcess = exec('node index.js', { cwd: botDir, env: configVar })
+    const botProcess = exec('NODE_ENV=test node index.js', { cwd: botDir, env: configVar }, err => {
+      checkTaskStatus(err)
+    })
     const CHECKING_BOT_STATUS_EVERY = 5000
 
     botProcess.stdout.on('data', data => {
@@ -128,12 +119,5 @@ describe('Create new bot', function() {
         done()
       }
     }, CHECKING_BOT_STATUS_EVERY)
-  })
-  /* eslint-disable-next-line */
-  after(() => {
-    const endTime = moment()
-    const testDuration = moment.duration(endTime.diff(startTime))
-
-    console.log('Test duration: ', testDuration.asMinutes(), 'm')
   })
 })

--- a/packages/core/botpress/tests/database_helpers.js
+++ b/packages/core/botpress/tests/database_helpers.js
@@ -39,7 +39,7 @@ run('helpers', () => {
   })
 
   describe('insertAndRetrieve', function() {
-    this.timeout(5000)
+    this.timeout(30000)
     itBoth('returns inserted data', (knex, sampleTable) => {
       return Promise.map(_.range(500), index => {
         const tString =

